### PR TITLE
[okina] MM methods to register non-okina pointers [artv3/okina-regPointers]

### LINKED ISSUE #783: [okina] Feature Request: External memory pointer tracking
For pointers not being tracked by the Okina memory manager, it would be great if we had a way to add them into the eco-system. 

Example:
```c++
double * ptr = new  double[10] 
mfem::Vector myVector(ptr, 10);
double *myVector_data =  Vector.GetData();
GET_PTR(myVector_data);
```

### PR DESCRIPTION
**Pull request by artv3**
Here's me taking a swing at issue #783 . This PR enabled the Okina memory manager to register external pointers; additionally this PR will return NULL device pointers if the host pointer is NULL.  

Resolves #783

### CODE REVIEW

File: general/mm.hpp [hunk 1]
```diff
@@ -112,6 +112,24 @@ class mm
...
+   {
+     MM().Insert(ptr_host, size*sizeof(T));
+   }
+
+   // **************************************************************************
+   template<class T>
+   static void RegisterHostAndDevicePtr(T * ptr_host, T * ptr_device, const size_t size)
+   {
+     MM().Insert(ptr_host, size*sizeof(T));
+     MM().maps.memories.at(ptr_host);
```
Comment (jdahm) [inline general/mm.hpp]: I don't think you need this line.

Comment (artv3) [reply general/mm.hpp]: Thanks! Good catch.

**Comment by artv3** [comment]
I'm giving this branch a spin on an application. It may need some adjustments...

**Comment by artv3** [comment]
Integrating external memory tracking through mfem::Vector constructors/methods which perform a shallow copy has been troubling. Namely due to the necessary garbage collection. I think we may have to re-evaluate how mfem::Vector gets used in applications. A possible solution is that any pointer that gets used within MFEM::Vector should first be registered with the memory manager and freed by the external application.

File: general/mm.hpp [hunk 2]
```diff
@@ -112,6 +112,23 @@ class mm
...
+   }
+
+   // **************************************************************************
+   template<class T>
+   static void RegisterHostAndDevicePtr(T * ptr_host, T * ptr_device, const size_t size)
+   {
+     MM().Insert(ptr_host, size*sizeof(T));
+     mm::memory &base = MM().maps.memories.at(ptr_host);
+     base.d_ptr = ptr_device;
+     base.host = false;
```
Comment (camierjs) [inline general/mm.hpp]: We should add the ability to the user to decide where the starting point is.
Setting this boolean to `false` means it is considered to be on the device, which might not be the case.

**Comment by artv3** [comment]
@camierjs , Yes! There are also examples in applications where folks insert memory allocated from the stack into an MFEM::Vector. With the previous iteration, I ran into cases where the data was getting registered multiple times.

**Pull request merged**